micro-rdk/src/common/data_manager.rs

```diff
@@ -27,6 +27,23 @@ use thiserror::Error;
 // the smaller amount of available RAM, we've halved it
 static MAX_SENSOR_CONTENTS_SIZE: usize = 32000;
 
+/// Allow for a C project using micro-RDK as a library to implement a callback to be run
+/// whenever data has successfully been uploaded. The callback should be identical in signature
+/// to the weak symbol declared here
+///
+/// Ex. for ESP-IDF project
+/// void data_upload_callback(void) {
+///     ESP_LOGI("proj_main", "hit data upload callback");
+/// }
+///
+/// # Safety
+/// The implemented callback should be simple and avoid using resources managed by
+/// micro-RDK. Long-running operations should be avoided.
```

Comment:
I still think there should be language here about disallowing reentrant calls and being careful with taking locks.